aspnet/mvc/controllers/routing.rst

```diff
@@ -1,8 +1,140 @@
+
+.. code-block:: c# 
+
+  app.UseMvc(routes =>
+  {
+    routes.MapRoute(
+      name: "default",
+      template: "{controller=Home}/{action=Index}/{id?}");
+  });
+  
+The route template defines the route as well as default values, in this example, ``Home`` for the controller, ``Index`` for the action method, and optionally ``id`` as a URL parameter. Routes defined in the ``Startup.cs`` file are conventional routes. 
+
+Conventional routing
+--------------------
+
+Routing first matches the URL to a route template, then selects a controller and action to route the request. You may define routes and route constraints on both controllers and actions. In conventional routing, you define a route that generates the action and controller tokens. The value of those tokens are used to select an action and controller by name. For example, the URL ending in ``/home/index/`` matches the route template for ``{controller}/{action}``. Conventional routing must be configured in the ``Startup`` class, as shown in the previous code sample. 
+
+For most forms-over-data web applications that use standard controllers, conventional routes work nicely. Attribute routing is the recommended type of routing for use in APIs.
+
+Attribute routing
+-----------------
+
+[Route],[Http*]
+^^^^^^^^^^^^^^^
+
+Rather having to rely on naming and conventions to figre out where to route the request, attribute routing enables you to use an attribute to route requests directly to the desired destination. You may apply attributes to either controllers or actions. Attribute routing is more direct because routes must match what is defined in the attribute in order to be processed. When an action method is decorated with the ``[Route]`` attribute, it can no longer be accessed through the convention-based routes, as the attribute takes precedence. 
+
+.. code-block:: c#
+
+  [Route("category/{movie}.rss", Order = 0, Name = "MovieCategoryFeed")]
+
+A common scenario for using attribute routing in an API controller is having the same route behave differently depending upon which HTTP verb is used. For example, you might want a page that displays a movie by its id using the route ``/movies/1337``, but also on that page be able to purchase the movie, and send the results back to the server. To restrict the HTTP verb that is accepted, apply the ``[HttpGet]`` attribute to the ``GetMovie`` action:
+
+.. code-block:: c#
+
+  [HttpGet("{id}", Name = "GetMovie")]
+  public IActionResult GetMovie([FromRoute] int id)
+
+The order in which routes are processed is important, but you can control how it works. First, routing inspects the ``RouteOrder`` property of the ``[Route]`` attribute. 
```

Comment:
What's missing here is that attribute routing builds a tree that will process your routes. By default you don't need to think about ordering because we can do something sensible, and the concept of "order" is rarely needed. Consider the following:
```
api/Products
api/Products/{id}
api/Products/{id}/Inventory
```

Based on the number of segments, we'll only process one of these.

I wouldn't dwell too much on the concept of order for attribute routes since it's rarely needed. We can generally do the right thing without you thinking about it.

> Routing matches routes first - then uses order when there's a tie - or when multiple match.

This isn't true. Routing builds a tree for each value of order defined and processes them in sequence.

===========

Where *Order* does matter is if you're doing something highly specialized. One common case is URL generation where an action has two routes and your want to prefer one of them for URLs that we generate.